topics/go/algorithms/strings/reverse/reverse.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+
+	fmt.Println(ReverseString("Hello World"))
+}
+
+func ReverseString(str string) string {
+
+	// Here we create a slice of codePoints.
+	codePoints := []rune(str)
+
+	// Here we create int that will be a pointer to the front of the codePoints.
+	var fontCodePoint int
+
+	// Here  we create int that will be a pointer to the end of the codePoints.
+	backCodePoint := len(codePoints) - 1
+
+	// While fontCodePoint is less than backCodePoint.
+	for fontCodePoint < backCodePoint {
+
+		// Here we swap the values of the slice at the codePoints[fontCodePoint] and codePoints[backCodePoint].
+		codePoints[fontCodePoint], codePoints[backCodePoint] = codePoints[backCodePoint], codePoints[fontCodePoint]
```

Comment:
This maybe easier to understand if you break this into two lines of code. It’s not obvious (at least to me) what’s happening because the line is so long and the variable names are long

Reply:
What if I used a temp variable like this:
```
		// Swap the code points.
		c := codePoints[beg]

		codePoints[beg] = codePoints[end]

		codePoints[end] = c
```

Reply:
Call it cp and I think I like it. Need to look at the larger code view. But remove the line spaces since those three lines work together.

Reply:
Using `r` for rune might make sense as well. I’ve been wondering is codepoint shouldn’t be rune. Not sure what’s more accurate or clear.